### PR TITLE
Update CanvasKit to 0.7.0 and flesh out painting

### PR DESCRIPTION
**Pull request by hterkelsen**
This allows us to fix some bugs in the CanvasKit backend.

- Implement RRect where the radii are different
- Implement drawDRRect
- Implement ColorFilter
- Implement the correct `arcTo` for `arcToPoint`

# Screenshots

## Using `drawDRRect` for checkboxes
drrect

## Showing `ColorFilter` in the "Animation" demo
colorfilter

## Using correct `arcTo` to get the correct notch under the FAB
correctnotch

### CODE REVIEW

File: lib/web_ui/lib/src/engine/compositor/color_filter.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      filter._color.value,
+      makeSkBlendMode(filter._blendMode),
+    ]);
+  }
+
+  SkColorFilter.matrix(EngineColorFilter filter) {
+    final js.JsArray colorMatrix = js.JsArray();
+    colorMatrix.length = 20;
+    for (int i = 0; i < 20; i++) {
+      colorMatrix[i] = filter._matrix[i];
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/compositor/color_filter.dart]: Let's leave a todo to remove this double-conversion. This should, at least, convert straight to a `Float64List`, but ideally we should write to WASM memory directly.

/cc @kjlubick - here's an example for excessive conversion that we talked about at the last sync.

File: lib/web_ui/lib/src/engine/compositor/initialization.dart [hunk 2]
```diff
@@ -6,10 +6,10 @@ part of engine;
 
 /// EXPERIMENTAL: Enable the Skia-based rendering backend.
 const bool experimentalUseSkia =
-    bool.fromEnvironment('FLUTTER_WEB_USE_SKIA', defaultValue: false);
+    bool.fromEnvironment('FLUTTER_WEB_USE_SKIA', defaultValue: true);
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/compositor/initialization.dart]: We probably want to keep this as `false`.

BTW, have you checked the code size difference with and without this backend? If the size difference is small (say <50KB gzipped), I'd say let's make it a dynamic option. I imagine most of the size cost is from the WASM blob, not from Dart code.

File: lib/web_ui/lib/src/engine/compositor/recording_canvas.dart [hunk 3]
```diff
@@ -114,7 +114,11 @@ class SkRecordingCanvas implements RecordingCanvas {
 
   @override
   void drawDRRect(ui.RRect outer, ui.RRect inner, ui.Paint paint) {
-    throw 'drawDRRect';
+    skCanvas.callMethod('drawDRRect', <js.JsObject>[
+      makeSkRRect(outer),
+      makeSkRRect(inner),
+      makeSkPaint(paint),
+    ]);
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/compositor/recording_canvas.dart]: 🎉

File: lib/web_ui/lib/src/engine/compositor/initialization.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/initialization.dart]: I'll investigate after landing this PR

File: lib/web_ui/lib/src/engine/compositor/color_filter.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/color_filter.dart]: Done.

**Pull request merged**